infra/src/main/kotlin/fi/fta/geoviite/infra/geometry/GeometryController.kt

```diff
@@ -150,6 +153,23 @@ class GeometryController @Autowired constructor(private val geometryService: Geo
         return geometryService.getElementListing(id, elementTypes)
     }
 
+    @PreAuthorize(AUTH_ALL_READ)
+    @GetMapping("/plans/{id}/element-listing/file")
+    fun getPlanElementListingCsv(
+        @PathVariable("id") id: IntId<GeometryPlan>,
+        @RequestParam("elementTypes") elementTypes: List<GeometryElementType>,
+    ): ResponseEntity<ByteArray> {
+        log.apiCall("getPlanElementList")
+        val (filename, content) = geometryService.getElementListingCsv(id, elementTypes)
+
+        val headers = HttpHeaders()
+        headers.contentType = MediaType.APPLICATION_OCTET_STREAM
+        headers.set(
+            HttpHeaders.CONTENT_DISPOSITION,
+            ContentDisposition.attachment().filename("${filename}.csv").build().toString())
+        return ResponseEntity.ok().headers(headers).body(content)
```

Comment:
Toteutettu samaan tapaan miten InfraModelien lataus oli tehty

Reply:
Nyt kun näitä on parissa paikkaa niin tuon responsen tekemisestä tiedostolle voisi tehdä funkkarin ResponseEntity.kt tiedostoon